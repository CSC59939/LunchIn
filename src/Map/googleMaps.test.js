import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import GoogleM from "./googleMaps";

describe("Maps Component", () => {
  describe("When provided with no info", () => {
    it("Should render correctly", () => {
      const mapInstance = shallow(
        <GoogleM />
      );
      expect(mapInstance).toMatchSnapshot();
    });
  });
  describe("When provided with food and address info", () => {
    it("Should render correctly", () => {
      const food = "Pizza";
      const address = "160 Convent Ave, New York, NY 10031"
      const mapInstance = shallow(
        <GoogleM 
            food={food} 
            address={address}
        />
      );
      expect(mapInstance).toMatchSnapshot();
    });
  });
});
