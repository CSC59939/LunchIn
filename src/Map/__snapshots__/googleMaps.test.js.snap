// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps Component When provided with food and address info Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMapComponent
    address="160 Convent Ave, New York, NY 10031"
    food="Pizza"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Card)>
        <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
          address="160 Convent Ave, New York, NY 10031"
          food="Pizza"
          lat={0}
          lng={0}
        />
      </WithStyles(Card)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
          address="160 Convent Ave, New York, NY 10031"
          food="Pizza"
          lat={0}
          lng={0}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "address": "160 Convent Ave, New York, NY 10031",
          "food": "Pizza",
          "lat": 0,
          "lng": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Card)>
          <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
            address="160 Convent Ave, New York, NY 10031"
            food="Pizza"
            lat={0}
            lng={0}
          />
        </WithStyles(Card)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withProps(withScriptjs(withGoogleMap(withState(withState(withState(withState(withHandlers(Component))))))))
            address="160 Convent Ave, New York, NY 10031"
            food="Pizza"
            lat={0}
            lng={0}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "address": "160 Convent Ave, New York, NY 10031",
            "food": "Pizza",
            "lat": 0,
            "lng": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Maps Component When provided with no info Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMapComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
